### ISSUE TITLE
MangaDex v3

### ISSUE THREAD

**Issue by Vesnyx** [DakuTree/manga-tracker#361]
Soon, MangaDex will be upgrading to their latest version, which includes a new reader and a semi-different paging system. A preview can be shown here: http://beta.mangadex.org/

It might be best to prepare a new display/tracking script beforehand.

**Comment by DakuTree** [comment]
From the looks of things, it looks the new site has a proper API now, and it works pretty great.

Example Title API: http://beta.mangadex.org/api/manga/23439
Example Chapter API :   http://beta.mangadex.org/api/chapter/404243

They also appear to have a usable JS API:

```
API.Chapter.create({id: 404243}).then(data => {
	console.log(data);
	/* This contains all chapter & manga data:
	   - manga
	     - chapterList (This is all chapter data according to language settings)
		 - chapters (This is all chapter data, regardless of language settings
		 - _data (Manga Data)
	   - _data (Current Chapter Data)
})
```

From what I can tell, this uses the data that has already been requested on the page, so no duplicate requests as well which is nice.

I'll have a look more at this tomorrow, but it _should_ be simple enough to add support for this.

**Comment by DakuTree** [comment]
Both the userscript & backend now have beta support.  

At the moment the only issue is the lack of MAL ID in the new API, which means we may have to switch back to the old method for grabbing chapter data at some point. I did request that it gets added on the forums but that may be a while before it happens (if it does).

**Comment by DakuTree** [comment]
Seems like they removed the JS API for some reason. The `API` var no longer exists, nor any similar ones.  
Did a temp-fix in e08e442b6fb728bd7deb568d144ebd671439ad03, but this does mean we're doing duplicate API requests which slows things down a fair bit.

**Comment by DakuTree** [comment]
v3 is now live as per https://mangadex.org/thread/20750.

Changes have been pushed on our end too - cda433ae83d32c4d70aba836c9930261f2df62b7

Sadly it still appears the public API is gone. Maybe it will come back at some point, but for now we need to double-request.
